spec.html

```diff
@@ -31878,23 +31882,21 @@ <h1>Assertion</h1>
             1. Assert: _x_ is a State.
             1. Assert: _c_ is a Continuation.
             1. Let _e_ be _x_'s _endIndex_.
-            1. Call IsWordChar(_e_ - 1) and let _a_ be the Boolean result.
-            1. Call IsWordChar(_e_) and let _b_ be the Boolean result.
-            1. If _a_ is *true* and _b_ is *false*, or if _a_ is *false* and _b_ is *true*, then
-              1. Call _c_(_x_) and return its result.
-            1. Return ~failure~.
+            1. Let _a_ be ! IsWordChar(_e_ - 1).
+            1. Let _b_ be ! IsWordChar(_e_).
+            1. If _a_ = _b_, return ~failure~.
```

Comment:
> `_a_ = _b_`

Since equality is a confusing concept in JavaScript, I would prefer never to use `=` to talk about anything except mathematical values. I prefer the previous, more explicit text.

I also prefer keeping this as `1. If foo, return _c_(_x_). 1. Return ~failure~.`, rather than the other way around, for consistency with the rest of this section.

Reply:
Your point about the ordering is fair, but I want to make the "do these tests return the same result?" logic underlying this algorithm more obvious... do you really think that `=` is hard to understand for Boolean values? And if so, how would you feel about using SameValue instead?

Reply:
> do you really think that `=` is hard to understand for Boolean values?

I think it requires a lot more thinking about it than the current wording, yes. Ditto for SameValue.

Reply:
I'm not going to die on this hill, but I would like some other opinions.

Reply:
cc @ljharb @michaelficarra @syg thoughts?

Reply:
In JavaScript, `=` means assignment, and it causes a lot of bugs when people assume it works for comparison. I think using `=` for comparison in the spec for JavaScript would be very confusing.

I'm totally fine with SameValue - I love using AOs, and I wish [`BinaryXor`](https://tc39.es/ecma262/#sec-binaryxor) was trivially usable here - but the original wording is also fine to me.

Reply:
Editor group as a whole is in favor of keeping the existing wording, though we all had slightly different reasons for it (and @michaelficarra was fine with your proposed wording).

Reply:
Ok, reverted.